bash_completion.sh

```diff
@@ -128,11 +128,32 @@ _buildtest ()
 {
   local cur="${COMP_WORDS[COMP_CWORD]}"
   local prev="${COMP_WORDS[COMP_CWORD-1]}"
+  declare -i offset="0"
 
   COMPREPLY=()   # Array variable storing the possible completions.
 
-  next=${COMP_WORDS[1]}
+  declare -a buildtest_opts=("--color" "--config" "--debug" "--editor" "--help" "--helpcolor" "--logpath" "--loglevel" "--print-log" "--no-color" "--report" "--version" "--view-log" "-c" "-d" "-h" "-l" "-p" "-r" "-V") 
+  
+  for command in "${COMP_WORDS[@]}"
+  do
+    for element in "${buildtest_opts[@]}"
+    do
+
+        if [[ "$command" == "$element" ]]; then
+          
+          if [[ "$command" == "--color" ]] || [[ "$command" == "--config" ]] || [[ "$command" == "-c" ]] || [[ "$command" == "--report" ]] || [[ "$command" == "-r" ]] || [[ "$command" == "--loglevel" ]] || [[ "$command" == "-l" ]] || [[ "$command" == "--editor" ]];
```

Comment:
this long if condition looks messy, i believe you can do a check for variable in a list of values something like this. I am not sure if this will work.


see https://techieroop.com/check-if-array-contains-value-in-shell-script/

I just got this code-snippet from chatGPT. 


```bash
commands=( "--color" "--config" "-c" "--report" "-r" "--loglevel" "-l" "--editor" )

if [[ "${commands[*]}" =~ "$command" ]]; then
  # code to be executed if command is found in the list
fi
```

Reply:
here is another rewrite of the code from chatGPT it even smaller though i am not sure if the array assignment make sense at first until you read the code-base later on

```bash
declare -A buildtest_opts=(
  ["--color"]=2
  ["--config"]=2
  ["--debug"]=1
  ["--editor"]=2
  ["--help"]=1
  ["--helpcolor"]=1
  ["--logpath"]=2
  ["--loglevel"]=2
  ["--print-log"]=1
  ["--no-color"]=1
  ["--report"]=2
  ["--version"]=1
  ["--view-log"]=1
  ["-c"]=2
  ["-d"]=1
  ["-h"]=1
  ["-l"]=2
  ["-p"]=1
  ["-r"]=2
  ["-V"]=1
)

offset=0
for command in "${COMP_WORDS[@]}"
do
  if [[ -n "${buildtest_opts[$command]}" ]]; then
    ((offset+=buildtest_opts[$command]))
  else
    ((offset+=1))
  fi
done
```

Reply:
@shahzebsiddiqui Yeah, I had tried with `declare -A` before, it is for associative arrays but it only works for bash version 4.

Reply:
Replacing the long if condition with a check for variable in a list works. 

<img width="1326" alt="Screen Shot 2023-04-11 at 1 10 05 PM" src="https://user-images.githubusercontent.com/35829130/231278020-c81f52f7-347b-4279-a6c0-ea3123214fde.png">